packages/compile/src/model/model.js

```diff
@@ -38,14 +38,45 @@ const PRINT_INIT_GRAPH = false
 const PRINT_AUX_GRAPH = false
 const PRINT_LEVEL_GRAPH = false
 
-function read(parseTree, spec, extData, directData, modelDirname) {
+// XXX: This is needed for tests due to variables being in module-level storage
+function resetModelState() {
+  variables.length = 0
+  inputVars.length = 0
+  variablesByName.clear()
+  constantExprs.clear()
+  nonAtoANames = Object.create(null)
+}
+
+/**
+ * Read the given model parse tree and resolve all subscript and variable/equation definitions.
+ *
+ * Note that this function currently does not return anything and instead stores the parsed subscript
+ * definitions in the `subscript` module and the parsed/analyzed variables in this module.
+ *
+ * @param {import('../parse/parser.js').VensimModelParseTree} parseTree The Vensim parse tree.
```

Comment:
This is an example of a place where I started defining some types via JSDoc.  I only added type info where it was straightforward, and for other cases I left it as `*` (wildcard / any).  I figure it's good enough for now (not really the main focus of this PR) and something we can improve over time.

Reply:
I like adding JSDoc for documentation purposes. If we get type checking out of it too, that's great. I haven't found much need for type checking in SDEverywhere development, as the data structures and class hierarchy aren't that complicated. I do forget what function arguments mean, though, so documenting them would be helpful to me.